### DISCUSSION TITLE
[SCENE] Scene de redirection automatique en fonction du support utilisé (HUB)

### DISCUSSION THREAD

**Discussion by WITNicolas** [witsa/synapps#510] [Show and tell]
Scene exportée du modèle OneView et developpé par @manufitoussiwit 

Faire démarrer la SYNAPP sur cette scene.
Choisir les scenes a afficher en fonction de la taille détectée
(par défaut 760px. Si > alors Gabarit Desktop. Si < Gabarit Smartphone

![image](https://user-images.githubusercontent.com/33632664/176474405-c696c97f-f7e4-4258-913f-eb3da4316123.png)

````text
SYNAPPS-STUDIO-SCENE|{"config":{"key":"hub","name":"Hub","additionalDefs":{"desktopMaster":{"type":"scene","label":"Gabarit Desktop","helperTxt":"Scène de Gabarit pour PC et tablette"},"smartphoneMaster":{"type":"scene","label":"Gabarit Smartphone","helperTxt":"Scène de Gabarit pour Smartphone"},"size":{"type":"number","label":"Taille limite","unit":"px"}},"additionals":{"desktopMaster":"master_D","smartphoneMaster":"master_S","size":760}},"leadActor":{"key":"stack1","type":"layout/stack","properties":{"verticalAlignment":"expand"},"events":{"onInit":["const smartphoneMaster = this.scene.additionals.smartphoneMaster;","const desktopMaster = this.scene.additionals.desktopMaster;","const size = this.scene.additionals.size || 760;","const isEmpty = context.utils.isEmpty;","if (isEmpty(smartphoneMaster) || isEmpty(smartphoneMaster)) return;","this._navTimeout = setTimeout(() => {","  if (this.isDestroyed || this.isDestroying) return;","  if (this.synapp.isInDesigner) return;","  var isSmallScreen = window.matchMedia(`only screen and (max-width: ${size}px), only screen and (max-height: ${size}px)`);","  var isTouchEnabled = 'ontouchstart' in window || navigator.maxTouchPoints > 0 || navigator.msMaxTouchPoints > 0;","  var hasMouse = Boolean(window.matchMedia('(hover: hover) and (pointer: fine)').matches);","  if (isSmallScreen.matches && isTouchEnabled && !hasMouse) {","    this.synapp.navigate(this.scene.additionals.smartphoneMaster);","  } else {","    this.synapp.navigate(this.scene.additionals.desktopMaster);","  }","}, 500);"],"onDestroy":["if (this._navTimeout) clearTimeout(this._navTimeout);"]}}}
````

**Comment by RaphaelE2IT** [comment]
Bonjour, 

Je suis dans une impasse, le code ne veut pas fonctionner lorsque je le copie/colle sur la zone de code. Il me retire le nom de la scène (laissant un vide et une erreur car pas de nom dans la clé = impossible d'identifier la scène). Il n'y a aucun conflit dans les noms de clé. Je suis aussi en version 1.4.1

Je vous envoie une capture d'écran pour vous montrer l'effet que cela fait.
![Capture d’écran 2025-05-26 154240](https://github.com/user-attachments/assets/c25330a2-8681-455a-a4e8-b8021c840af0)

En espérant que vous trouverez des réponses à mon problème

Société E2IT


Edit : J'ai résolu mon problème en réinstallant Synapps, mon anti virus avait décidé de s'amuser avec le dossier source du logiciel.
